lib/helpers.js

```diff
@@ -35,6 +35,25 @@ var helpers = module.exports = {
       }
     }
 
+    // The following 2 if statements prevent a bug that was turning
+    // arrays into objects in the output.
+    // We now check if an array is only in the child or parent, return that.
```

Comment:
Hi @ShirajG and @prolificeric I find the following statements confusing. I'd appreciate if some insight could be provided. First of all, why are the inner conditions `_.every(collection), _isObject`? All expressions are evaluated and only the value from the last one is return and, as a result, the condition always evaluates to true. Wasn't `_.every(collection, _isObject)` the intended expression?. Additionally, when `_.merge(a,b)` is returned the first condition is `_.isArray(b) && !a`, doesn't this mean that when the return statement is reached an array is always merged with a falsy value?